Jenkinsfile

```diff
@@ -0,0 +1,324 @@
+    /*     toxenvs: ['py27', 'py34'], */
+    /* ], */
+    /* [ */
+    /*     label: 'docker', */
+    /*     imageName: 'pyca/cryptography-runner-sid', */
+    /*     toxenvs: ['py27', 'py35'], */
+    /* ], */
+    /* [ */
+    /*     label: 'docker', */
+    /*     imageName: 'pyca/cryptography-runner-stretch', */
+    /*     toxenvs: ['py27', 'py35'], */
+    /* ], */
+    /* [ */
+    /*     label: 'docker', */
+    /*     imageName: 'pyca/cryptography-runner-jessie-libressl:2.4.5', */
+    /*     toxenvs: ['py27'], */
+    /* ], */
+    /* [ */
+    /*     label: 'docker', */
+    /*     imageName: 'pyca/cryptography-runner-jessie-libressl:2.5.3', */
+    /*     toxenvs: ['py27'], */
+    /* ], */
+    /* [ */
+    /*     label: 'docker', */
+    /*     imageName: 'pyca/cryptography-runner-ubuntu-xenial', */
+    /*     toxenvs: ['py27', 'py35'], */
+    /* ], */
+    /* [ */
+    /*     label: 'docker', */
+    /*     imageName: 'pyca/cryptography-runner-ubuntu-rolling', */
+    /*     toxenvs: ['py27', 'py35', 'docs', 'pep8', 'py3pep8'], */
+    /* ], */
+    [
+        label: 'docker',
+        imageName: 'pyca/cryptography-runner-fedora',
+        toxenvs: ['py27', 'py35'],
+    ],
+]
+
+def downstreams = [
+    [
+        downstreamName: 'pyOpenSSL',
+        label: 'docker',
+        imageName: 'pyca/cryptography-runner-ubuntu-rolling',
+        script: """#!/bin/bash -xe
+            git clone --depth=1 https://github.com/pyca/pyopenssl.git pyopenssl
+            cd pyopenssl
+            virtualenv .venv
+            source .venv/bin/activate
+            pip install ../cryptography
+            pip install -e .
+            pip install pytest
+            pytest tests
+        """
+    ],
+    [
+        downstreamName: 'Twisted',
+        label: 'docker',
+        imageName: 'pyca/cryptography-runner-ubuntu-rolling',
+        script: """#!/bin/bash -xe
+            git clone --depth=1 https://github.com/twisted/twisted.git twisted
+            cd twisted
+            virtualenv .venv
+            source .venv/bin/activate
+            pip install ../cryptography
+            pip install pyopenssl service_identity pycrypto
+            pip install -e .
+            python -m twisted.trial src/twisted
+        """
+    ],
+    [
+        downstreamName: 'paramiko',
+        label: 'docker',
+        imageName: 'pyca/cryptography-runner-ubuntu-rolling',
+        script: """#!/bin/bash -xe
+            git clone --depth=1 https://github.com/paramiko/paramiko.git paramiko
+            cd paramiko
+            virtualenv .venv
+            source .venv/bin/activate
+            pip install ../cryptography
+            pip install -e .
+            pip install -r dev-requirements.txt
+            inv test
+        """
+    ],
+]
+
+def checkout_git(label) {
+    def script = ""
+    if (env.BRANCH_NAME.startsWith('PR-')) {
+        script = """
+        git clone --depth=1 https://github.com/pyca/cryptography.git cryptography
```

Comment:
Is it pratical to factor these first two lines of the script that are the same in both things out?